lib/hooks.js

```diff
@@ -1,30 +1,41 @@
 const RiffError = require('./riff-error');
 const {types: errorTypes} = require('./errors');
 
-const handleError = (reject) => (err) => {
-    return reject(new RiffError(errorTypes.HOOK_RUNTIME, err));
+const handleError = (err) => {
+    if (err instanceof RiffError) {
+        return err
+    }
+    return new RiffError(errorTypes.HOOK_RUNTIME, err);
 };
 
-module.exports.guardWithTimeout = (hookFn, timeoutInMs) => {
-    return new Promise((resolve, reject) => {
-        const hookType = typeof hookFn;
-        if (hookType === 'undefined') {
-            return resolve();
-        }
-        if (hookType !== 'function') {
-            return reject(new RiffError(errorTypes.HOOK_INVALID, `Hooks must be functions, found: ${hookType}`));
-        }
+const invalidHookError = (hookType) => {
+    return new RiffError(errorTypes.HOOK_INVALID, `Hooks must be functions, found: ${hookType}`);
+};
+
+const timeoutError = (timeoutInMs) => {
+    return new RiffError(errorTypes.HOOK_TIMEOUT, `The hook took too long to run (timeout: ${timeoutInMs}ms). Aborting now`);
+};
+
+module.exports.guardWithTimeout = async (hookFn, timeoutInMs) => {
+    const hookType = typeof hookFn;
+    if (hookType === 'undefined') {
+        return;
+    }
+    if (hookType !== 'function') {
+        throw invalidHookError(hookType);
+    }
 
-        const timerId = setTimeout(() => {
-            reject(new RiffError(errorTypes.HOOK_TIMEOUT, `The hook took too long to run (timeout: ${timeoutInMs}ms). Aborting now`));
-        }, timeoutInMs);
 
-        try {
-            Promise.resolve(hookFn())
-                .then(resolve, handleError(reject))
-                .finally(() => clearTimeout(timerId));
-        } catch (err) {
-            handleError(reject)(err);
-        }
-    });
+    try {
+        let timerId;
+        const hookPromise = Promise.resolve(hookFn());
```

Comment:
If you apply this, then the try block and be unindented and the catch block can be completely removed.

```suggestion
        const hookPromise = Promise.resolve().then(hookFn);
```